iothub_client/inc/iothub_client_properties.h

```diff
@@ -0,0 +1,272 @@
+
+/* @brief Enumeration that indicates whether the JSON value of a deserialized property 
+*         is a null-terminated string or binary.
+*/
+typedef enum IOTHUB_CLIENT_PROPERTY_VALUE_TYPE_TAG
+{
+    /* @brief Deserialized JSON value is a string. */
+    IOTHUB_CLIENT_PROPERTY_VALUE_STRING,
+    /* @brief Deserialized JSON value is a binary. */
+    IOTHUB_CLIENT_PROPERTY_VALUE_BINARY
+} IOTHUB_CLIENT_PROPERTY_VALUE_TYPE;
+
+/** @brief Current version of @p IOTHUB_CLIENT_DESERIALIZED_PROPERTY structure.  */
+#define IOTHUB_CLIENT_DESERIALIZED_PROPERTY_VERSION_1 1
+
+/** @brief    This struct represents a property from IoT Hub.  
+              It is generated by IoTHubClient_Deserialize_Properties while deserializing a property payload. */
+typedef struct IOTHUB_CLIENT_DESERIALIZED_PROPERTY_TAG {
+    /** @brief   Version of the structure.  Currently will be set to IOTHUB_CLIENT_DESERIALIZED_PROPERTY_VERSION_1 by IoTHubClient_Serialize_ReportedProperties. */
+    int structVersion;
+    /** @brief   Whether this is a property the device reported (and we're seeing what IoT Hub last received from the device)
+         or whether it's a writable property request from the device. */
+    IOTHUB_CLIENT_PROPERTY_TYPE propertyType;
+    /** @brief Name of the component.  Optional; may be NULL for the root component. */
+    const char* componentName;
+    /** @brief Name of the property. */
+    const char* name;
+    /** @brief Whether the value is passed as a string or non-null terminated length. */
+    /** @description  Applications can safely assume that this will always be IOTHUB_CLIENT_PROPERTY_VALUE_STRING.  
+                      They do not need to write code to handle IOTHUB_CLIENT_PROPERTY_VALUE_BINARY.
+                      IOTHUB_CLIENT_PROPERTY_VALUE_BINARY may be supported by the SDK in the future, but IOTHUB_CLIENT_PROPERTY_VALUE_STRING will *always* by the default.
+    */
+    IOTHUB_CLIENT_PROPERTY_VALUE_TYPE valueType;
+    /** @brief Value of the property. Currently this is only set to str. */
+    union {
+        const char* str;
+        unsigned const char* bin;
+    } value;
+    /** @brief Number of bytes in valueLength.  Does not include terminating \0 if IOTHUB_CLIENT_PROPERTY_VALUE_STRING is set */
+    size_t valueLength;
+} IOTHUB_CLIENT_DESERIALIZED_PROPERTY;
+
+/**
+* @brief   Serializes reported properties into a format for sending to IoT Hub.
+*
+* @param[in]   properties                  Pointer to IOTHUB_CLIENT_REPORTED_PROPERTY to be serialized.
+* @param[in]   numProperties               Number of elements contained in @p properties.
+* @param[in]   componentName               Optional component name these properties are part of.  May be NULL for default component.
+* @param[out]  serializedProperties        Serialized output of @p properties for sending to IoT Hub.
+                                           The application must release this memory using IoTHubClient_SerializedProperties_Destroy.
```

Comment:
Typo - `IoTHubClient_SerializedProperties_Destroy` doesn't exist, should have been `IoTHubClient_Serialize_Properties_Destroy`.  I'll correct in next round.